cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1058,4 +1141,22 @@ public boolean apply(Privilege input) {
     Assert.assertTrue(Sets.filter(authorizer.listPrivileges(ALICE), entityFilter).isEmpty());
     Assert.assertTrue(Sets.filter(authorizer.listPrivileges(BOB), entityFilter).isEmpty());
   }
+
+  private void verifyDatasetIsEmpty(NamespaceId namespaceId, String datasetName) throws Exception {
+    DataSetManager<KeyValueTable> outTableManager = getDataset(namespaceId.toId(), datasetName);
+    KeyValueTable outputTable = outTableManager.get();
+    Assert.assertFalse(outputTable.scan(null, null).hasNext());
+  }
+
+  private void verifyProgramFailure(Map<String, String> programArgs, ProgramManager programManager)
+    throws TimeoutException, InterruptedException {
+    programManager.start(programArgs);
+    programManager.waitForFinish(5, TimeUnit.MINUTES);
```

Comment:
On that note, can you report a jira to add this run record check as part of ``waitForFinish``